app/elements/coder/profile/profile-page.html

```diff
@@ -119,7 +126,13 @@ <h3 class="card-header">Teams</h3>
 				</template>
 			</paper-material>
 			<paper-material class="right" elevation="1">
-				<h3 class="card-header">Challenges</h3>
+				<template is="dom-if" if="[[isCurrentUser]]">
```

Comment:
What if you remove the `dom-if` and just put `<user-challenges-table>` ? It's data provider will only recieve the visible challenges for the currently logged in user from the platform.